modules/standard/IO.chpl

```diff
@@ -3751,6 +3751,12 @@ record binarySerializer {
 
         This method is only optimized for the case where the
         ``binarySerializer`` has been configured for ``native`` endianness.
+
+      .. warning::
+
+        This method should only be called when the ``data`` argument is located
+        on the same locale as the underlying ``writer`` of this serializer.
```

Comment:
I thought the issue came up when the data is on a different locale than the `file`, not the `fileWriter`?

Reply:
Ah yes, thanks for catching that. Fixed!